docs/directives.md

```diff
@@ -0,0 +1,55 @@
+# @glint Directives
+
+Glint has several directives that can be used in template comments to control
+Glint's behavior. Additional comment text can follow directives to document
+their purpose.
+
+## @glint-ignore
+
+The `@glint-ignore` directive tells Glint to ignore the line that follows it.
+Glint will not report any errors encountered on the next line.
+
+Example:
+
+```hbs
+<MyComponent @expectedArg="foo" />
+
+{{! @glint-ignore: need to update signature to accept this arg }}
+<MyComponent @unexpectedArg="bar" />
+```
+
+## @glint-expect-error
+
+The `@glint-expect-error` directive operates similarly to `@glint-ignore` in
+that it will not report type errors it encounters, but it will also produce an
+error when an error is _not_ encountered. This is useful for tests where we
+expect an invocation not to type-check (e.g. due to bad arguments) and want to
+be alerted if it does.
```

Comment:
My inclination (though if you and/or @chriskrycho disagree I'm happy to defer!) would be to introduce `expect-error` first as the more general error-squashing solution

```suggestion
be alerted if it does, or if you know something currently won't typecheck due
to an issue elsewhere and want to know when it's been fixed and you can
remove the directive.
```

Then in the docs for `ignore`, we'd explicitly call out that `ignore` is more of a special-case thing that you typically wouldn't reach for.

Reply:
Yep, I agree here!